pkg/kube/pod.go

```diff
@@ -133,3 +144,58 @@ func WaitForPodCompletion(ctx context.Context, cli kubernetes.Interface, namespa
 	})
 	return errors.Wrap(err, "Pod did not transition into complete state")
 }
+
+// PatchDefaultPodSpecs use Strategic Merge to patch default pod specs with the passed specs
+func PatchDefaultPodSpecs(defaultPodSpecs v1.PodSpec, override sp.JSONMap) (v1.PodSpec, error) {
```

Comment:
(From meeting discussion) Lets avoid exporting  func `PatchDefaultPodSpecs ` and  `CreateAndMergeJsonPatch`

Reply:
We'll still need `CreateAndMergeJsonPatch` I believe

Reply:
Yes, `CreateAndMergeJsonPatch` is being used in `function` package